src/openforms/submissions/tests/test_exports.py

```diff
@@ -167,3 +172,112 @@ def test_user_defined_variables_in_export(self):
                 "Some value",
             ),
         )
+
+    @freeze_time("2022-05-09T13:00:00Z")
+    def test_submission_export_with_mixed_fields_and_forms(self):
```

Comment:
Can you add a regression test that reproduces the reported issue _without_ mixed forms? That's the case that primarily needs to be solved. I'm a bit on the fence of the practicality of exporting mixed forms - this requires a lot of upfront knowledge to the importing party and they can't tell if a key belongs to the form in the row or not based on whether the value is empty or not.